crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -110,32 +113,70 @@ impl RuffSettingsIndex {
         // Add any settings within the workspace itself.
         for directory in WalkDir::new(root)
             .into_iter()
+            .filter_entry(|entry: &DirEntry| {
+                if !entry.file_type().is_dir() {
+                    return true;
+                }
+
+                let directory = entry.path();
+
+                // If the directory is excluded from the workspace, skip it.
+                if let Some((_, settings)) = index
+                    .borrow()
+                    .range(..directory.to_path_buf())
+                    .rev()
+                    .find(|(path, _)| directory.starts_with(path))
+                {
+                    if let Some(file_name) = directory.file_name() {
+                        let candidate = Candidate::new(&directory);
+                        let basename = Candidate::new(file_name);
+                        if match_candidate_exclusion(
+                            &candidate,
+                            &basename,
+                            &settings.file_resolver.exclude,
+                        ) {
+                            tracing::debug!("Ignored path via `exclude`: {}", directory.display());
+                            return false;
+                        } else if match_candidate_exclusion(
+                            &candidate,
+                            &basename,
+                            &settings.file_resolver.extend_exclude,
+                        ) {
+                            tracing::debug!(
+                                "Ignored path via `extend-exclude`: {}",
+                                directory.display()
+                            );
+                            return false;
+                        }
+                    }
+                }
+
+                true
+            })
```

Comment:
I first assumed that the `{` starts a new block, and only after removing it realised that it starts the `for` body. 

I think it would be good to extract the `WalkDir::new` into a variable